File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/doctor/controller/dto/RegisterDoctorForm.java
package org.back.systemklinikimedycznej.doctor.controller.dto;

import org.back.systemklinikimedycznej.personal_details.dto.PersonalDetailsForm;
import org.back.systemklinikimedycznej.user.dto.RegisterAccountForm;

import java.time.LocalDate;
import java.util.List;

public record RegisterDoctorForm(
        RegisterAccountForm registerAccountData,
        PersonalDetailsForm personalDetails,
        String pwzNumber,
        LocalDate dateOfEmployment,
        List<DoctorSpecializationDto> doctorSpecializations
){
}

File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/doctor/domain/register-doctor-form.ts
import {Account} from '../../account/model/account';
import {PersonalDetails} from '../../personal-details/domain/personal-details';
import {DoctorSpecialization} from './doctor-specialization';

export interface RegisterDoctorForm {
  registerAccountData: Account;
  personalDetails: PersonalDetails;
  pwzNumber: string;
  description: string;
  dateOfEmployment: string;
  doctorSpecializations: DoctorSpecialization[];
}
